core/ecschema-metadata/src/test/Metadata/Class.test.ts

```diff
@@ -99,6 +100,181 @@ describe("ECClass", () => {
     });
   });
 
+  describe("adding and deleting properties from classes", () => {
+    let entityClass: EntityClass;
+
+    beforeEach(() => {
+      schema = new Schema(new SchemaContext(), "TestSchema", "ts", 1, 0, 0);
+      entityClass = new EntityClass(schema, "TestClass");
+    });
+
+    it("should do nothing when deleting property name that is not in class", async () => {
+      expect(entityClass.properties).to.be.undefined;
+      expect(await entityClass.getProperty("TestProp", true)).to.be.undefined;
+
+      await (entityClass as ECClass as MutableClass).deleteProperty("TestProp");
+
+      expect(entityClass.properties).to.be.undefined;
+      expect(await entityClass.getProperty("TestProp", true)).to.be.undefined;
+    });
+
+    it("should do nothing when deleting property name that is not in class, synchronous", async () => {
+      expect(entityClass.properties).to.be.undefined;
+      expect(await entityClass.getProperty("TestProp", true)).to.be.undefined;
+
+      (entityClass as ECClass as MutableClass).deletePropertySync("TestProp");
+
+      expect(entityClass.properties).to.be.undefined;
+      expect(await entityClass.getProperty("TestProp", true)).to.be.undefined;
+    });
+
+    it("should do nothing if a property is already deleted", async () => {
```

Comment:
Think this test and below's `deleteProperty` and `deletePropertySync` is flipped.